hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/infrastructure/HapiApiClients.java

```diff
@@ -117,12 +126,13 @@ private void ensureChannelStubsInPool(final NodeConnectInfo node, final String u
             final var channel = createNettyChannel(useTls, node.getHost(), node.getPort(), node.getTlsPort());
```

Comment:
A question on line 116 above (GitHub doesn't allow me to comment on it): why is there an `if` and not a `while`?
The very first spec will have only one connection created - a limitation difficult to justify.

Reply:
I'm not sure why this was written the way it was (I'm just changing it to be round-robin).  I think it's initializing channels as needed rather than creating the whole pool at once (most specs only need a small number of channels).
This method is called in a stream forEach down below.

Reply:
@tinker-michaelj ?

Reply:
@jsync-swirlds is correct, I was creating at most one channel per initialized `HapiSpec`. I think I still don't have great intuition about what has changed recently to make channel selection such a sensitive topic. 🤔

Reply:
GRPC has bugs in how the *client* handles multiple messages on a single connection.  Moving to more connections and avoiding concurrent reuse in the integration test client resolves the "integration tests hang forever" issues in the netty-grpc code.